### PR TITLE
Fix a set of serialization and API issues for ArcsInstant 

### PR DESCRIPTION
**Pull request by Cypher1**
Fixes:
- Add missing imports for ArcsInstant and ArcsDuration to code generation for Particles and Plans
- Fix early parse failure in refinement and paxel numeric value in type annotation and unit (e.g. 12days would be parsed as 12d)
- Improve conformance of non-jvm ArcsInstant and ArcsDuration APIs to jvm base
- Fix up conversion of ArcsInstant to ReferencablePrimitive (add missing parameter)
- Implement missing case for fromLiteral for DiscretePrimitiveNode
- Improve error for refinement failure in StorageAdapter (Provide Schema information in error)

### CODE REVIEW

**Comment by csilvestrini** [comment]
> _No description provided._

Would be good to list in the PR description was the issues were that you're fixing

File: java/arcs/core/util/platform/native/PlatformInstant.kt [hunk 1]
```diff
@@ -18,9 +18,11 @@ class PlatformInstant {
     fun toEpochMilli(): Long =
         TODO("Add support for ArcsInstant in Kotlin Native")
 
+    @Suppress("UNUSED_PARAMETER")
     fun compareTo(other: PlatformInstant): Int =
         TODO("Add support for ArcsInstant in Kotlin Native")
 
+    @Suppress("UNUSED_PARAMETER")
```
Comment (csilvestrini) [inline java/arcs/core/util/platform/native/PlatformInstant.kt:25]: This one seems weird. You are using the `other` parameter, so why is this needed?

File: java/arcs/core/util/platform/native/PlatformDuration.kt [hunk 2]
```diff
@@ -37,5 +38,9 @@ class PlatformDuration {
         @Suppress("UNUSED_PARAMETER")
         fun ofDays(days: Long): PlatformDuration =
             TODO("Add support for ArcsDuration in Kotlin Native")
+
+        @Suppress("UNUSED_PARAMETER")
+        fun ofHours(hours: Long): PlatformDuration =
+            TODO("Add support for ArcsDuration in Kotlin Native")
```
Comment (csilvestrini) [inline java/arcs/core/util/platform/native/PlatformDuration.kt:44]: Would be great to have a tracking bug for these, you can include it in the TODO

File: src/runtime/manifest-parser.pegjs [hunk 3]
```diff
@@ -1972,9 +1969,12 @@ UnitName
     return unit+'s';
   }
 
+TypeIdentifier = typeIdentifier:('n'/'i'/'l'/'f'/'d') &[^a-zA-Z0-9_] {
```
Comment (csilvestrini) [inline src/runtime/manifest-parser.pegjs:1972]: Could you add a comment for this one? I don't really understand what it is.

**Review by csilvestrini — APPROVED** [review]

File: java/arcs/core/util/platform/native/PlatformInstant.kt (hunk 1 above)
Comment (Cypher1) [reply java/arcs/core/util/platform/native/PlatformInstant.kt:25]: Ah, it's not! Removing it

File: src/runtime/manifest-parser.pegjs (hunk 3 above)
Comment (Cypher1) [reply src/runtime/manifest-parser.pegjs]: Updated, any feedback?

Comment (csilvestrini) [reply src/runtime/manifest-parser.pegjs]: That helps, thanks!

**Comment by Cypher1** [comment]
The breakage seen here should be fixed by https://github.com/PolymerLabs/arcs/pull/6183

File: java/arcs/core/util/platform/native/PlatformDuration.kt (hunk 2 above)
Comment (Cypher1) [reply java/arcs/core/util/platform/native/PlatformDuration.kt]: Filed https://b.corp.google.com/issues/169213588 to track adding support for BigInt, ArcsInstant and ArcsDuration to js+native.

Hopefully they're only needed temporarily as Kotlin comes along.

**Review by csilvestrini — APPROVED** [review]

**Comment by Cypher1** [comment]
Pushing again after rebase (hoping to see tests pass internally this time)
